crates/burn-autodiff/src/lib.rs

```diff
@@ -34,3 +35,15 @@ pub use backend::*;
 
 #[cfg(feature = "export_tests")]
 mod tests;
+
+/// A facade around for HashMap and HashSet.
+/// This avoids elaborate import wrangling having to happen in every module.
+mod collections {
+    cfg_block::cfg_block! {
+        if #[cfg(feature = "std")] {
+            pub use std::collections::{HashMap, HashSet};
+        } else {
+            pub use hashbrown::{HashMap, HashSet};
+        }
+    }
```

Comment:
Since there is only one line now for collections, we can remove the cfg_block dependency and use

```suggestion
    #[cfg(feature = "std")]
    pub use std::collections::{HashMap, HashSet};
    #[cfg(not(feature = "std"))]
    pub use hashbrown::{HashMap, HashSet};
```